### PR TITLE
Update ResourceBounds to use spec.Uint128

### PR DESCRIPTION
**Pull request by RuneRogue**
Fixing #2237

### CODE REVIEW

**Comment by RuneRogue** [comment]
I need your help . I was trying hard to solve some error but still I was not able to solve this error .
`Received unexpected error:
            	            	json: cannot unmarshal string into Go struct field ResourceBounds.transaction.resource_bounds.max_price_per_unit of type spec.Uint128`
            	            	
 I am getting this error during the test cover .

File: core/transaction.go [hunk 1]
```diff
@@ -57,18 +58,20 @@ const (
...
 	// MaxPricePerUnit is technically a uint128
-	MaxPricePerUnit *felt.Felt
+	MaxPricePerUnit *spec.Uint128
 }
 
 func (rb ResourceBounds) Bytes(resource Resource) []byte {
 	maxAmountBytes := make([]byte, 8) //nolint:mnd
 	binary.BigEndian.PutUint64(maxAmountBytes, rb.MaxAmount)
-	maxPriceBytes := rb.MaxPricePerUnit.Bytes()
+	maxPriceBytes := make([]byte, 16) //nolint:mnd
```
Comment (derrix060) [inline core/transaction.go:67]: I've seen this logic before (https://github.com/NethermindEth/juno/pull/2242/files#diff-8fcd72f4c5e1f056a6875680f66dfbfc8578908ff4fe889c8f1fc21e94e9c8bbR157) please DRY and create a function to do these steps, properly unittested :)

Comment (RuneRogue) [reply core/transaction.go]: Made changes

**Comment by RuneRogue** [comment]
> Screenshot 2024-10-27 203256 I need your help . I was trying hard to solve some error but still I was not able to solve this error . I am getting this error during the test .

want your help me to solve this issue .

**Comment by pnowosie** [comment]
What is a reason to use p2p/spec/Uint128 in rpc and even worse in the core modules?

I'm against this change, because it puts additional hurdle to client libraries that now would need to handle non-standard integer type. Also p2p/spec/Uimt128 is protobuf-specific which should not be used in rpc imo.

**Pull request closed without merging**